src/tests/Interop/UnmanagedCallersOnly/UnmanagedCallersOnlyTest.cs

```diff
@@ -225,9 +225,9 @@ public static void TestUnmanagedCallersOnlyWithGeneric()
 
         Assert.Equal(0, ((delegate* unmanaged<MaybeBlittable<nint>, int>)&MaybeBlittableGenericStruct)(new MaybeBlittable<nint>()));
 
-        Assert.Throws<InvalidProgramException> (() => ((delegate* unmanaged<NotBlittable<int>, int>)&InvalidGenericUnmanagedCallersOnlyParameters.GenericClass)(new NotBlittable<int>()));
+        Assert.Throws<MarshalDirectiveException> (() => ((delegate* unmanaged<NotBlittable<int>, int>)&InvalidGenericUnmanagedCallersOnlyParameters.GenericClass)(new NotBlittable<int>()));
```

Comment:
Can we change this to use the "wrong" type on the managed side (like `nint`) to ensure that we get to the UnmanagedCallersOnly callee? The `MarshalDirectiveException` is on the managed->unmanaged side.